Extension/src/LanguageServer/client.ts

```diff
@@ -1883,11 +1879,11 @@
             pendingTask = new util.BlockingTask<T>(task, pendingTask);
             return pendingTask.getPromise();
         } else {
-            return Promise.reject(localize("unsupported.client", "Unsupported client"));
+            throw new Error (localize("unsupported.client", "Unsupported client"));
```

Comment:
Do you need to mark this function as `async` to ensure that this throw is converted to a promise rejection?